### PR TITLE
#57 -  Header

### PR DESCRIPTION
**Pull request by jhancock532**
[Ticket on Codebase](https://projects.torchbox.com/projects/contentful-careers-page/tickets/57).

This WIP draft contains the mobile header and navigation menu. I've refactored the HTML and SCSS from the main website, adding interactivity and page scroll locking using React with `body-scroll-lock`.

### CODE REVIEW

File: components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+type HeaderProps = {
+    jobCount?: number;
+    theme: Theme;
+};
+
+export const Header = ({ theme, jobCount = 8 }: HeaderProps) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const mobileNavRef = useRef<HTMLDivElement>(null);
+
+    function toggleMobileMenu() {
```
Comment (Morsey187) [inline components/Header.tsx:46]: **Non-blocking** personally I prefer arrow functions

File: components/TorchboxLogo.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+type TorchboxLogoProps = {
+    theme: Theme;
+    className?: string;
+};
+
+export const TorchboxLogo = ({ theme, className }: TorchboxLogoProps) => (
+    <svg
+        className={className}
+        xmlns="http://www.w3.org/2000/svg"
+        width="137"
```
Comment (Morsey187) [inline components/TorchboxLogo.tsx:12]: **Though:** I tend to add [viewbox](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/viewBox) attributes manually  when they're missing, they can make it easier to size SVGs if needed in the future.

**Review by Morsey187 — APPROVED** [review]
Looks good to me, just a few non-blocking comments

File: components/ChevronIcon.tsx [hunk 3]
```diff
@@ -15,6 +15,6 @@ export const ChevronIcon = ({ className }: ChevronIconProps) => (
             fillRule="evenodd"
             d="M4.854 0L0 4.206 4.708 10 0 15.794 4.854 20l8.006-9.853.14-.122-.02-.025.02-.025-.14-.122L4.854 0z"
             clipRule="evenodd"
-        ></path>
+        />
```
[the remark this answers is not in the document]
Comment (jhancock532) [reply components/ChevronIcon.tsx:18]: That's good to know, in this case React will automatically add the self closing tag when it adds this element to the DOM.

File: components/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        url: 'https://torchbox.com/team/',
+    },
+];
+
+type HeaderProps = {
+    jobCount?: number;
+    theme: Theme;
+};
+
+export const Header = ({ theme, jobCount = 8 }: HeaderProps) => {
```
[the remark this answers is not in the document]
Comment (jhancock532) [reply components/Header.tsx:42]: Thank you, I will change it to zero and a placeholder in the layout

File: components/Header.tsx (hunk 1 above)
Comment (jhancock532) [reply components/Header.tsx:46]: I've been using arrows elsewhere, good to keep it consistent

File: components/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        } else {
+            setIsOpen(true);
+            disableBodyScroll(mobileNav);
+        }
+    };
+
+    const themeClass = theme === 'DARK' ? 'themeDark' : 'themeLight';
+
+    return (
+        <div className={[styles.container, themeClass].join(' ')}>
```
Comment (jhancock532) [inline components/Header.tsx:62]: Set theme in the Layout component, not here: theme should be accessible everywhere in the document.

File: components/DesktopNav.tsx [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+);
+
+type SubnavProps = {
+    links: NavLink[];
+    jobsAvailable?: number;
+    toggleMenu: () => void;
+};
+
+const Subnav = ({ links, jobsAvailable = 0, toggleMenu }: SubnavProps) => {
+    // Use a ref as we don't need to rerender the component on tab navigation
```
Comment (jhancock532) [inline components/DesktopNav.tsx:40]: Create a distinct file for the Subnav component, this one is too large.

File: components/Header.tsx (hunk 5 above)
Comment (Morsey187) [reply components/Header.tsx:62]: I think you could just do string concatination here `${styles.container} ${themeClass}`, I remember suggesting this before but I feel like that was related to an already existing array of string values. These variables look to already be strings so creating an array shouldn't be neccassary.

You could also look into a useContext hook for keeping the theme value in sync.

File: README.md [hunk 7]
```diff
@@ -36,7 +35,10 @@ Log in to Contentful and from Settings > API Keys populate `.env.local` with the
 Restart the development server.
 
 To view the GraphQL API explorer, go to
-`https://graphql.contentful.com/content/v1/spaces/{ SPACE ID }/environments/staging/explore?access_token={ ACCESS TOKEN }`
+
+```
+https://graphql.contentful.com/content/v1/spaces/{ SPACE ID }/environments/staging/explore?access_token={ ACCESS TOKEN }
```
Comment (Morsey187) [inline README.md:40]: 👍

**Review by Morsey187 — APPROVED** [review]
LGTM!

File: components/Header.tsx (hunk 5 above)
Comment (jhancock532) [reply components/Header.tsx:62]: I've been swapping between different forms of contactination in this project, I think I will add the above suggestion for two classes throughout my project, and use the join for three or more classes.

Comment (jhancock532) [reply components/Header.tsx:62]: UseContext for the theme value looks interesting, I will have to learn more about this. As we have SCSS modules I think it's nicer to keep all the styling within the SCSS files, so we don't have to update the classes or styles conditionally with React.

File: components/DesktopNav.tsx [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+        .slice(0, 3)
+        .map((link, index) => <DesktopNavItem link={link} key={index} />);
+
+    const modalIsOpenClass = isOpen ? styles.desktopNav__modalOpen : '';
+
+    return (
+        <div className={styles.desktopNav}>
+            <div className={styles.desktopNav__container}>
+                <nav
+                    className={`${styles.desktopNav__modal} ${modalIsOpenClass}`}
```
[the remark this answers is not in the document]
Comment (jhancock532) [reply components/DesktopNav.tsx:176]: Cool, I was wondering about this one.

**Pull request merged**